common/gcpURLParts.go

```diff
@@ -17,13 +17,13 @@ type GCPURLParts struct {
 	UnparsedParams string
 }
 
-const gcpHostPattern = "^storage.cloud.google.com"
+const gcpHostPattern = "^storage\\.cloud\\.google\\.com"
```

Comment:
More then codeQL this appears to be a bug in regex filter. How is this thing working in current code?

Reply:
It passes more than necessary through, so this regex will pass normal google hosts and more.